### ISSUE TITLE
refetchQueries only refetches a single query

### ISSUE THREAD

**Issue by wmertens** [apollographql/react-apollo#1897]
**Intended outcome:**

After a mutation, the `refetchQueries` prop is used and all queries that have matching names are refetched.

**Actual outcome:**

After a mutation, the `refetchQueries` prop is used and the first query for each matching name is refetched.

**How to reproduce the issue:**

I don't have time to create a reproduction right now, I'm hoping this rings a bell, if not I can create a repro.

Basically, we have several queries with the same name that have slightly different fields, eg. `query foo { foo { id bar } }` and `query foo { foo { id meep } }`. Since v2.1 of react-apollo, only one of those queries gets updated.

**Version**

* apollo-client@2.2.8
* react-apollo@2.1.2

It used to work before.

**Comment by wmertens** [comment]
I should add that changing the name of the queries and adding it to the refetchQueries works around the problem, but I do believe it should not be necessary.

**Comment by hwillson** [comment]
React Apollo has been refactored to use React Hooks behind the scenes for everything, which means a lot has changed since this issue was opened (and a lot of outstanding issues have been resolved). We'll close this issue off, but please let us know if you're still encountering this problem using React Apollo >= 3.1.0. Thanks!
